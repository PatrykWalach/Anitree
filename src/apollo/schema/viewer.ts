export interface User {
  id: number
  name: string
  avata: UserAvatar
  siteUrl: string
  options: {
    profileColor: string
    displayAdultContent: boolean
  }
  mediaListOptions: {
    scoreFormat: ScoreFormat
    animeList: MediaListTypeOptions
    mangaList: MediaListTypeOptions
  }
}
export interface UserAvatar {
  large: string
  medium: string
}
export type ScoreFormat =
  | 'POINT_100'
  | 'POINT_10_DECIMAL'
  | 'POINT_10'
  | 'POINT_5'
  | 'POINT_3'

export interface MediaListTypeOptions {
  customLists: string[]
  advancedScoring: string[]
  advancedScoringEnabled: boolean
}
