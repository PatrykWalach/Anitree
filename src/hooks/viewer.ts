import { inject } from '@vue/composition-api'
import { useQuery } from '@vue/apollo-composable'
export const DefaultViewer = Symbol('Viewer')

export const useViewer = <V = ReturnType<typeof useQuery>>() =>
  inject(DefaultViewer) as V
