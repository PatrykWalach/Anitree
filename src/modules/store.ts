import * as modules from '.'
import { Store, plugin } from 'vuex-composition-api'

export const store =
  process.env.NODE_ENV === 'production'
    ? undefined
    : new Store({
        plugins: [plugin(Object.values(modules))],
      })
