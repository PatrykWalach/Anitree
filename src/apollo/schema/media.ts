import { ScoreFormat } from './viewer'
import { MediaList } from './mediaListCollection'

export interface Variables {
  isAdult?: boolean
  id: number
  format?: ScoreFormat
}

export interface Node {
  id: number
  type: MediaType
}

export type MediaType = 'MANGA' | 'ANIME'
export interface FuzzyDate {
  year: number | null
  month: number | null
  day: number | null
}

export interface Media extends Node {
  startDate: FuzzyDate
  endDate: FuzzyDate
  format: string | null
  description: string | null
  season: string | null
  seasonInt: number | null
  episodes: number | null
  chapters: number | null
  volumes: number | null
  isFavourite: boolean
  mediaListEntry: MediaList | null
  title: MediaTitle
  siteUrl: string
  relations: {
    edges: MediaEdge[]
  }
  tags: {
    rank: number
    name: string
  }[]
  studios: {
    nodes: Studio[]
  }
  status: MediaStatus | null
  bannerImage: string | null
  coverImage: MediaCoverImage
  stats: MediaStats
}
export interface MediaStats {
  scoreDistribution: ScoreDistribution[] | null
}
export interface ScoreDistribution {
  score: number
  amount: number
}
export interface MediaTitle {
  romaji: string
  english: string
  native: string
}

export type MediaRelation =
  | 'ADAPTATION'
  | 'PREQUEL'
  | 'SEQUEL'
  | 'PARENT'
  | 'SIDE_STORY'
  | 'CHARACTER'
  | 'SUMMARY'
  | 'ALTERNATIVE'
  | 'SPIN_OFF'
  | 'OTHER'
  | 'SOURCE'
  | 'COMPILATION'
  | 'CONTAINS'

export interface MediaEdge {
  node: Node
  relationType: MediaRelation
  id: number
}

export interface Studio {
  id: number
  name: string
}

export interface MediaCoverImage {
  extraLarge: string | null
  large: string | null
  medium: string | null
  color: string | null
}

export type MediaStatus =
  | 'CURRENT'
  | 'PLANNING'
  | 'COMPLETED'
  | 'DROPPED'
  | 'PAUSED'
  | 'REPEATING'
