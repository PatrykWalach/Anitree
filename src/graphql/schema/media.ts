import { ScoreFormat } from './viewer'
import { MediaList } from './mediaListCollection'

export interface Variables {
  isAdult?: boolean
  id: number
  format?: ScoreFormat
}

export interface Node {
  __typename: 'Media'
  id: number
  type: MediaType
}

export type MediaType = 'MANGA' | 'ANIME'
export interface FuzzyDate {
  __typename: 'FuzzyDate'
  year: number | null
  month: number | null
  day: number | null
}

export interface Media extends Node {
  startDate: FuzzyDate
  endDate: FuzzyDate
  format: string | null
  description: string | null
  season: string | null
  seasonInt: number | null
  episodes: number | null
  chapters: number | null
  volumes: number | null
  isFavourite: boolean
  ranknigs: MediaRank[]
  mediaListEntry: MediaList | null
  title: MediaTitle
  siteUrl: string
  relations: MediaConnection
  tags: MediaTag[]
  studios: StudioConnection
  status: MediaStatus | null
  bannerImage: string | null
  coverImage: MediaCoverImage
  stats: MediaStats
}
export interface MediaConnection {
  __typename: 'MediaConnection'
  edges: MediaEdge[]
}
export interface MediaTag {
  __typename: 'MediaTag'
  id: number
  rank: number
  name: string
}
export interface StudioConnection {
  __typename: 'StudioConnection'

  nodes: Studio[]
}
export interface MediaRank {
  __typename: 'MediaRank'
  allTime: boolean
  context: string
  rank: number
  type: 'RATED' | 'POPULAR'
  year: null | number
}
export interface MediaStats {
  __typename: 'MediaStats'
  scoreDistribution: ScoreDistribution[] | null
}
export interface ScoreDistribution {
  __typename: 'ScoreDistribution'
  score: number
  amount: number
}
export interface MediaTitle {
  __typename: 'MediaTitle'
  romaji: string
  english: string
  native: string
}

export type MediaRelation =
  | 'ADAPTATION'
  | 'PREQUEL'
  | 'SEQUEL'
  | 'PARENT'
  | 'SIDE_STORY'
  | 'CHARACTER'
  | 'SUMMARY'
  | 'ALTERNATIVE'
  | 'SPIN_OFF'
  | 'OTHER'
  | 'SOURCE'
  | 'COMPILATION'
  | 'CONTAINS'

export interface MediaEdge {
  __typename: 'MediaEdge'
  node: Node
  relationType: MediaRelation
  id: number
}

export interface Studio {
  __typename: 'Studio'
  id: number
  name: string
}

export interface MediaCoverImage {
  __typename: 'MediaCoverImage'
  extraLarge: string | null
  large: string | null
  medium: string | null
  color: string | null
}

export type MediaStatus =
  | 'CURRENT'
  | 'PLANNING'
  | 'COMPLETED'
  | 'DROPPED'
  | 'PAUSED'
  | 'REPEATING'
