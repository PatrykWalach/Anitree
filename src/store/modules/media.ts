import Vue from 'vue'
import { apolloQueryPage } from '../api'
import {
  Module,
  VuexModule,
  Mutation,
  Action,
  getModule,
  MutationAction
} from 'vuex-module-decorators'
import store from '@/store'
import {
  MediaNode,
  Media,
  MediaEdgeExtended,
  MediaEdge,
  Data,
  RawFilter,
  Filter,
  FetchVariables,
  Page
} from '@/types'

function createVariables(
  acc: {
    idIn: number[]
  }[],
  media: MediaNode
): {
  idIn: number[]
}[] {
  const index = acc.findIndex(arr => arr.idIn.length < 50)
  if (index === -1) {
    acc.push({ idIn: [media.id] })
  } else {
    acc[index].idIn.push(media.id)
  }
  return acc
}

function applyFilter(
  filteredEdges: MediaEdgeExtended[],
  filter: RawFilter
): MediaEdgeExtended[] {
  return filteredEdges.filter(filter)
}

function filterDuplicate<E = any>(
  test: (element: E, otherElement: E, i?: number, arr?: E[]) => boolean,
  element: E,
  i: number,
  arr: E[]
): boolean {
  return arr.findIndex(test.bind(null, element)) === i
}

function nodesAreEqual(element: MediaNode, otherElement: MediaNode) {
  return element.id === otherElement.id
}

function filterDuplicateNodes(
  node: MediaNode,
  i: number,
  allNodes: MediaNode[]
): boolean {
  return filterDuplicate(nodesAreEqual, node, i, allNodes)
}

function filterDuplicateEdges(
  edge: MediaEdge,
  i: number,
  allEdges: MediaEdge[]
): boolean {
  return filterDuplicateNodes(edge.node, i, allEdges.map(({ node }) => node))
}
@Module({
  namespaced: true,
  name: 'media',
  store,
  dynamic: true
})
export class ModuleMedia extends VuexModule {
  public currentId: number | null = null

  public media: Data<Media> = {}

  public filters: Data<Filter> = {}

  // public filteredMedia: MediaEdgeExtended[] = []

  public get activeFilters() {
    return Object.values(this.filters).filter(({ active }) => active)
  }

  public get exclusiveFilters(): RawFilter[] {
    return this.activeFilters
      .filter(({ exclusive }) => exclusive)
      .map(({ filter }) => filter)
  }

  public get inclusiveFilters(): RawFilter[] {
    return this.activeFilters
      .filter(({ exclusive }) => !exclusive)
      .map(({ filter }) => filter)
  }

  public get currentMedia(): Media | null {
    const { currentId, media } = this
    return (currentId && media[currentId]) || null
  }

  public get relationTypes(): string[] {
    const { inclusiveFilters, filteredMedia } = this
    return [
      ...new Set(
        inclusiveFilters
          .reduce(applyFilter, filteredMedia)
          .flatMap(media => media.node.relations.edges)
          .map(({ relationType }) => relationType)
      )
    ]
  }

  public get sortedMedia(): Media[] {
    const getYear = (seasonInt: number) => {
      const year = Math.floor(seasonInt / 10)
      return year > 50 ? 1900 + year : 2000 + year
    }

    return this.filteredMedia
      .map(({ node }) => node)
      .sort(
        (
          { startDate: dateA, seasonInt: seasonA },
          { startDate: dateB, seasonInt: seasonB }
        ) =>
          (seasonA &&
            seasonB &&
            (getYear(seasonA) - getYear(seasonB) ||
              (seasonA % 10) - (seasonB % 10))) ||
          (dateA.year && dateB.year && dateA.year - dateB.year) ||
          (dateA.month && dateB.month && dateA.month - dateB.month) ||
          (dateA.day && dateB.day && dateA.day - dateB.day) ||
          1
      )
  }

  public get filteredMedia(): MediaEdgeExtended[] {
    const getRelatedMedia = (
      edges: MediaEdge[],
      acc: MediaEdgeExtended[] = []
    ): MediaEdgeExtended[] => {
      const { exclusiveFilters, inclusiveFilters, media } = this

      const newMedia = exclusiveFilters.reduce(
        applyFilter,
        edges
          .map(({ relationType, node }) => {
            return {
              relationType,
              node: media[node.id]
            }
          })
          .filter(({ node }) => node)
      )

      acc.push(...newMedia)
      const accNodes = acc.map(({ node }) => node)
      const relatedNodes = inclusiveFilters
        .reduce(applyFilter, newMedia)
        .map(({ node }) => node)
        .flatMap(({ relations }) => relations.edges)
        .filter(filterDuplicateEdges)
        .filter(({ node }) => !accNodes.some(nodesAreEqual.bind(null, node)))

      if (relatedNodes.length) {
        return getRelatedMedia(relatedNodes, acc)
      }

      return acc
    }

    const { currentId } = this

    return currentId
      ? getRelatedMedia([
          {
            node: {
              id: currentId,
              type: ''
            },
            relationType: '',
            id: -1
          }
        ])
      : []
  }

  @Mutation
  public CHANGE_FILTER(filter: Filter): Filter {
    return Vue.set(this.filters, filter.id, filter)
  }

  @Mutation
  public ADD_MEDIA(media: Media): Media {
    return Vue.set(this.media, media.id, media)
  }

  @MutationAction
  public async CHANGE_CURRENT_ID({
    currentId
  }: {
    currentId: number | null
  }): Promise<{
    currentId: number | null
  }> {
    return { currentId }
  }

  @Action
  public async extendMediaEdge({
    relationType,
    node
  }: MediaEdge): Promise<MediaEdgeExtended> {
    return Promise.resolve({
      relationType,
      node: await this.getMedia(node.id)
    })
  }

  @Action
  public async fetchMedia(variables: FetchVariables): Promise<Media[]> {
    return this.fetchMediaPage(variables).then(page => page.media)
  }

  @Action
  public async fetchMediaPage(variables: FetchVariables): Promise<Page> {
    return new Promise(resolve =>
      apolloQueryPage(variables)
        .then(page => {
          const newMedia = page.media
          newMedia.forEach(this.ADD_MEDIA)
          return page
        })
        .then(resolve)
        .catch(() =>
          setTimeout(() => this.fetchMediaPage(variables).then(resolve), 10000)
        )
    )
  }

  @Action
  public async handleQueue(medias: Media[]): Promise<void> {
    return new Promise(resolve => {
      const { media } = this
      const newMedias: MediaNode[] = medias
        .flatMap(newMedia => newMedia.relations.edges)
        .map(({ node }) => node)
        .filter(filterDuplicateNodes)
        .filter(newMedia => !media[newMedia.id])

      if (newMedias.length) {
        const { fetchMedia, handleQueue } = this
        Promise.all(newMedias.reduce(createVariables, []).map(fetchMedia))
          .then(result => result.flat())
          .then(handleQueue)
          .then(resolve)
      } else {
        resolve()
      }
    })
  }

  // @Action
  // public async getFilteredMedia({
  //   edges = [
  //     {
  //       node: {
  //         id: this.currentId || -1,
  //         type: ''
  //       },
  //       relationType: '',
  //       id: -1
  //     }
  //   ],
  //   acc = []
  // }: {
  //   edges?: MediaEdge[]
  //   acc?: MediaEdgeExtended[]
  // }): Promise<MediaEdgeExtended[]> {
  //   return new Promise(async resolve => {
  //     const {
  //       exclusiveFilters,
  //       inclusiveFilters
  //       // ,media
  //     } = this

  //     const newMedia = exclusiveFilters.reduce(
  //       applyFilter,
  //       await Promise.all(edges.map(this.extendMediaEdge))
  //       // edges
  //       //   .map(({ relationType, node }) => {
  //       //     return {
  //       //       relationType,
  //       //       node: media[node.id]
  //       //     }
  //       //   })
  //       //   .filter(({ node }) => node)
  //     )

  //     acc.push(...newMedia)
  //     const accNodes = acc.map(({ node }) => node)
  //     const relatedNodes = inclusiveFilters
  //       .reduce(applyFilter, newMedia)
  //       .map(({ node }) => node)
  //       .flatMap(({ relations }) => relations.edges)
  //       .filter(filterDuplicateEdges)
  //       .filter(({ node }) => !accNodes.some(nodesAreEqual.bind(null, node)))

  //     if (relatedNodes.length) {
  //       this.getFilteredMedia({ edges: relatedNodes, acc }).then(resolve)
  //     } else {
  //       resolve(acc)
  //     }
  //   })
  // }

  // @Mutation
  // public CHANGE_FILTERED_MEDIA(filteredMedia: MediaEdgeExtended[]) {
  //   this.filteredMedia = filteredMedia
  // }

  // @Action({ commit: 'CHANGE_FILTERED_MEDIA' })
  // public async changeFilteredMedia() {
  //   return await this.getFilteredMedia({})
  // }

  @Action
  public async getMedia(id: number) {
    return new Promise<Media>(async (resolve, reject) => {
      const storedMedia = this.media[id]
      if (storedMedia) {
        resolve(storedMedia)
      } else {
        const media = await this.fetchMedia({
          idIn: [id]
        })

        if (media.length) {
          resolve(media.shift())
        } else {
          reject()
        }
      }
    })
  }
}
export const media =getModule(ModuleMedia)
export default media
