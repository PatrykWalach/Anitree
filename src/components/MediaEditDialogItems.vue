<template>
  <v-tabs-items v-model="syncedTab">
    <v-tab-item v-for="i in 4" :key="i" :value="'edit' + i">
      <v-card-text>
        <MediaEditDialogItemsTab
          :method="'edit' + i"
          v-bind="{
            user,
            media,
            form,
            scoreFormat,
            advancedScoring,
            manga,
            changeForm,
          }"
        />
      </v-card-text>
    </v-tab-item>
  </v-tabs-items>
</template>
<script lang="ts">
import { computed, createComponent } from '@vue/composition-api'
import { Media } from '@/graphql/schema/media'
import MediaEditDialogItemsTab from './MediaEditDialogItemsTab.vue'
import { State } from '@/store'
import { User } from '@/graphql/schema/viewer'
import { mediaListToForm } from '../graphql'
import { useSelector } from 'vue-redux-hooks'
import { SaveVariables } from '../graphql/schema/listEntry'

export interface Props {
  user: User
  media: Media
  tab: string
  changeForm({ form }: { form: Partial<SaveVariables> }): void
}

export default createComponent<Readonly<Props>>({
  components: {
    MediaEditDialogItemsTab,
  },
  props: {
    changeForm: { default: null, required: true, type: Function },
    media: { default: null, required: true, type: Object },
    tab: { default: 'edit1', required: true, type: String },
    user: { default: null, required: true, type: Object },
  },
  setup(props, { emit }) {
    const syncedTab = computed({
      get: () => props.tab,
      set: e => emit('update:tab', e),
    })
    const manga = computed(() => props.media.type === 'MANGA')

    const advancedScoring = computed(
      () =>
        props.user.mediaListOptions[manga.value ? 'mangaList' : 'animeList']
          .advancedScoring,
    )

    const scoreFormat = computed(() => {
      const [, max, round] = props.user.mediaListOptions.scoreFormat.split('_')

      return {
        max: parseInt(max),
        round: round === 'DECIMAL' ? 1 : 0,
      }
    })

    const stored = computed(() =>
      mediaListToForm(
        props.media.mediaListEntry,
        advancedScoring.value.slice(),
      ),
    )

    const editForm = useSelector((state: State) => state.edit.form)

    const form = computed(() => ({
      ...stored.value,
      ...editForm.value,
    }))

    return { advancedScoring, form, manga, scoreFormat, syncedTab }
  },
})
</script>
