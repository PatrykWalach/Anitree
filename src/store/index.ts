import Vue from 'vue'
import Vuex from 'vuex'

// import { getModule } from 'vuex-module-decorators'
Vue.use(Vuex)

import { ModuleMedia } from './modules/media'
import { ModuleTheme } from './modules/theme'
import { ModuleTitle } from './modules/title'
import { ModuleAuth } from './modules/auth'

export interface RootState {
  media: ModuleMedia
  theme: ModuleTheme
  title: ModuleTitle
  auth: ModuleAuth
}

export default new Vuex.Store<RootState>({})
