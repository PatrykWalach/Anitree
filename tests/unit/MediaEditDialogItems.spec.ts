import { createVariations, matchSnapshot } from './utils'
import MediaEditDialogItems from '@/components/MediaEditDialogItems.vue'
import { mockedMediaAndListEntry } from './mocks/media'
import { mockedViewer } from './mocks/viewer'
import { useMockedStore } from './mocks/store'

describe('MediaEditDialogItems', () => {
  createVariations({
    propsData: {
      tab: () => Array.from({ length: 4 }, (v, k) => 'edit' + (k + 1)),
    },
  }).forEach(settings =>
    matchSnapshot(MediaEditDialogItems, {
      propsData: {
        ...settings.propsData,
        media: mockedMediaAndListEntry,
        user: mockedViewer,
      },
      provide: { ...useMockedStore() },
    }),
  )
})
