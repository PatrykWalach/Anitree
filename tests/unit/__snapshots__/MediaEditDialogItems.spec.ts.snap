// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaEditDialogItems matches snapshot 1`] = `
<v-tabs-items-stub
  activeclass="v-window-item--active"
  nexticon="$next"
  previcon="$prev"
  value="edit1"
>
  <v-tab-item-stub
    value="edit1"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit1"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit2"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit2"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit3"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit3"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit4"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit4"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
</v-tabs-items-stub>
`;

exports[`MediaEditDialogItems matches snapshot 2`] = `
<v-tabs-items-stub
  activeclass="v-window-item--active"
  nexticon="$next"
  previcon="$prev"
  value="edit2"
>
  <v-tab-item-stub
    value="edit1"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit1"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit2"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit2"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit3"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit3"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit4"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit4"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
</v-tabs-items-stub>
`;

exports[`MediaEditDialogItems matches snapshot 3`] = `
<v-tabs-items-stub
  activeclass="v-window-item--active"
  nexticon="$next"
  previcon="$prev"
  value="edit3"
>
  <v-tab-item-stub
    value="edit1"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit1"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit2"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit2"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit3"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit3"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit4"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit4"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
</v-tabs-items-stub>
`;

exports[`MediaEditDialogItems matches snapshot 4`] = `
<v-tabs-items-stub
  activeclass="v-window-item--active"
  nexticon="$next"
  previcon="$prev"
  value="edit4"
>
  <v-tab-item-stub
    value="edit1"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit1"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit2"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit2"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit3"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit3"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
  <v-tab-item-stub
    value="edit4"
  >
    <v-card-text-stub>
      <mediaeditdialogitemstab-stub
        advancedscoring="Story,Characters,Visuals,Audio"
        form="[object Object]"
        media="[object Object]"
        method="edit4"
        scoreformat="[object Object]"
        user="[object Object]"
      />
    </v-card-text-stub>
  </v-tab-item-stub>
</v-tabs-items-stub>
`;
